File: src/problems/leetcode-cn/1653. 使字符串平衡的最少删除次数/solution/python/solution.py
class Solution:
    def minimumDeletions(self, s: str) -> int:
        length = len(s)
        if length < 2:
            return 0

        bc, delete = 0, 0
        
        for v in s:
            if v == 'b':
                bc += 1
            else:
                delete += 1
                if delete > bc:
                    delete = bc

        return delete

File: src/problems/leetcode-cn/1653. 使字符串平衡的最少删除次数/solution/go/solution.go
func minimumDeletions(s string) int {
    length := len(s)
    if length < 2 {
        return 0
    }

    bc, del := 0, 0 // [这次待删除b的次数, 上次最小操作次数]

    for i := 0; i < length; i++ {
        if s[i] == 'a' {
            // del = min(del+1, bc)
            del++
            if del > bc {
                del = bc
            }
        } else {
            bc++
        }
    }

    return del
}